features/step_definitions/update_tests_file_steps.rb

```diff
@@ -1,29 +1,118 @@
-Then('the .tests file should contain the properly formatted test') do |test_params|
-    expected = <<~TEST_SPEC
-    /*
-    @name: #{test_params['name']}
-    @points: #{test_params['points']}
-    @type: #{test_params['type']}
-    @target: #{test_params['target']}
-    */
-    <test>
-    #{test_params['test_code']}
-    </test>
-    TEST_SPEC
-
-    expect(File.read(
-        File.join(ENV["ASSIGNMENTS_BASE_PATH"],
-        @assignment.repository_name,
-        "code.tests"
-    ))).to include(expected)
+require 'rspec/mocks'
+
+Before do
+  RSpec::Mocks.setup
+  allow_any_instance_of(Assignment).to receive(:push_changes_to_github)
+  allow_any_instance_of(TestsHelper).to receive(:current_user_and_token).and_return(
+  [ User.create!(id: 1, name: 'Test User', email: 'test@example.com'), 'fake_github_token' ])
+end
+
+After do
+  RSpec::Mocks.teardown
+  FileUtils.rm_rf(ENV["ASSIGNMENTS_BASE_PATH"])
+end
+
+
+Then('the .tests file should contain the properly formatted test') do |table|
+  table.hashes.each do |test_params|
+    expected = "/*\n" +
+    "@name: #{test_params['name']}\n" +
+    "@points: #{test_params['points'].to_f}\n" +
+    "@test_type: #{test_params['type']}\n" +
+    "@target: #{test_params['target']}\n" +
+    "*/\n" +
+    "<test>\n" +
+    "#{test_params['test_code']}\n" +
+    "</test>\n\n"
+
+    file_path = File.join(ENV["ASSIGNMENTS_BASE_PATH"], @assignment.repository_name, "tests", "c++", "code.tests")
+    actual_content = File.read(file_path)
+    # expect the strings to be equAL
+    expect(actual_content).to include(expected)
+  end
 end
 
 Then('the .tests file should contain the remaining {string} tests') do |string|
-    # count the number of tests in the file
-    file = File.read(
-        File.join(ENV["ASSIGNMENTS_BASE_PATH"],
-        @assignment.repository_name,
-        "code.tests"
-    )).to include(expected)
-    expect(file.scan(/<test>/).count).to eq(string.to_i)
+  file_path = File.join(ENV["ASSIGNMENTS_BASE_PATH"], @assignment.repository_name, "tests", "c++", "code.tests")
+  file = File.read(file_path)
+  expect(file.scan(/<test>/).count).to eq(string.to_i)
+end
+
+When('I add a new unit test called {string}') do |string|
+  steps %(
+        When I create a new test with type "unit"
+        And with the name "#{string}"
+  )
+end
+
+When('I set it to {string} points') do |string|
+  steps %(
+        When with the points "#{string}"
+  )
+end
+
+When('I set the target to {string}') do |string|
+  steps %(
+        When with the target "#{string}"
+  )
+end
+
+When('I fill in the test block with {string}') do |string|
+  fill_in 'Actual test', with: string
+end
+
+Then('I should see a success message') do
+  expect(page).to have_content('Test was successfully')
```

Comment:
Shouldn't the message say "Test was successfully created"

Reply:
There are multiple messages that say `Test was successfully <action>". I'll see if I can replace with a regex to make it more clear.